src/server/routes/actions.ts

```diff
@@ -52,14 +53,37 @@ const dripRequestHandler = async (requestOpts: DripRequestType): Promise<DripRes
   }
 };
 
-router.post<unknown, DripResponse, DripRequestType>("/bot-endpoint", (req, res) => {
-  dripRequestHandler(req.body)
-    .then(res.send)
-    .catch((e) => {
+if (config.Get("EXTERNAL_ACCESS")) {
+  logger.info("The faucet is externally accessible for drip requests.");
+  router.post<unknown, DripResponse, DripRequestType>("/drip", async (req, res) => {
```

Comment:
why not having one endpoint for both bot & web?
then inside of the handler, you'd have clear way to validate, depends on EXTERNAL_ACCESS, it's either `false` 1) username 2) address, or `true` 1) IP 2) address
currently, when we use 2 separate api endpoints, the validation is a bit spread b/w the api controller and dripRequestHandler. Ideally, if we use dripRequestHandler in both apis, they should accept 2 clear contracts, depends on the EXTERNAL_ACCESS state.